lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+      format: 'Boolean',
+      env: 'CMD_CSP_ENABLE',
+      default: true
+    },
+    directives: {
+      doc: 'Additional CSP directives to the default ones required for CodiMD to work.',
+      default: {}
+    },
+    addDefaults: {
+      doc: 'Option to control whenever the recommended CSP directives should be add.',
+      format: 'Boolean',
+      default: true
+    },
+    addDisqus: {
+      doc: 'Option to disable the addition of Disqus rules to CSP. Disabling this option will break the Disqus integration.',
+      format: 'Boolean',
+      default: true
+    },
+    addGoogleAnalytics: {
+      doc: 'Option to disable the addition of Google Analytics rules to CSP. Disabling this option will break the Google Analytics integration.',
+      format: 'Boolean',
+      default: true
+    },
+    upgradeInsecureRequests: {
+      default: 'auto'
+    },
+    reportURI: {
+      doc: 'Allows to add a URL for CSP reports in case of violations.',
+      format: '*',
+      default: undefined
+    }
+  },
+  protocolUseSSL: {
+    doc: 'Set to use TLS protocol for resources path (only applied when domain is set).',
+    format: 'Boolean',
+    env: 'CMD_PROTOCOL_USESSL',
+    default: false
+  },
+  useCDN: {
+    doc: 'Set to use CDN resources or not',
+    format: 'Boolean',
+    env: 'CMD_USECDN',
+    default: true
+  },
+  allowAnonymous: {
+    doc: 'Set to allow anonymous usage. (Allow guests general usage of CodiMD.)',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS',
+    default: true
+  },
+  allowAnonymousEdits: {
+    doc: 'If `allowAnonymous` is `true`: allow users to select `freely` permission, allowing guests to edit existing notes.',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_ANONYMOUS_EDITS',
+    default: false
+  },
+  allowFreeURL: {
+    doc: 'Set to allow new note creation by accessing a nonexistent note URL. This is the behavior familiar from [Etherpad](https://github.com/ether/etherpad-lite).',
+    format: 'Boolean',
+    env: 'CMD_ALLOW_FREEURL',
+    default: false
+  },
+  forbiddenNoteIDs: {
+    doc: 'List of note IDs that can\'t be used. Main purpose is to prevent bugs due usage of paths in free-url-mode.',
+    env: 'CMD_FORBIDDEN_NOTE_IDS',
+    default: ['robots.txt', 'favicon.ico', 'api', 'build', 'css', 'docs', 'fonts', 'js', 'uploads', 'vendor', 'views']
+  },
+  defaultPermission: {
+    doc: 'Set notes default permission (only applied on signed-in users).',
+    format: function (val) {
+      if (!/(freely|editable|limited|locked|protected|private)/.test(val)) {
+        throw new Error('defaultPermission has to be set to `freely`, `editable`, `limited`, `locked`, `protected`, or `private`')
+      }
+    },
+    env: 'CMD_DEFAULT_PERMISSION',
+    default: 'editable'
+  },
+  dbURL: {
+    doc: 'Database URL',
+    format: '*',
+    default: undefined,
+    sensitive: true
+  },
+  db: {
+    host: {
+      doc: 'Database host name/IP',
+      format: '*',
```

Comment:
wouldn't using the domain regex make some sende? (this would at least exclude white spaces). same goes for `dbUrl` a few lines above.

Reply:
This actually should become a custom format in future. The problem is that it could be undefined and the domain name or similar would simply break that.

I guess this will be implemented with an own format properly towards the end of the week.